rsc/src/main/java/com/cloudera/livy/rsc/rpc/RpcServer.java

```diff
@@ -61,18 +63,74 @@
   private static final SecureRandom RND = new SecureRandom();
 
   private final String address;
-  private final Channel channel;
+  private  Channel channel;
   private final EventLoopGroup group;
   private final int port;
   private final ConcurrentMap<String, ClientInfo> pendingClients;
   private final RSCConf config;
-
+  private final String portRange;
+  private static enum PortRangeSchema{START_PORT, END_PORT, max};
+  private final String PORT_DELIMITER="~";
+  /**
+   * Creating RPC Server
+   * @param lconf
+   * @throws IOException
+   * @throws InterruptedException
+   */
   public RpcServer(RSCConf lconf) throws IOException, InterruptedException {
     this.config = lconf;
+    this.portRange=config.get(LAUNCHER_PORT_RANGE);
     this.group = new NioEventLoopGroup(
-        this.config.getInt(RPC_MAX_THREADS),
-        Utils.newDaemonThreadFactory("RPC-Handler-%d"));
-    this.channel = new ServerBootstrap()
+    this.config.getInt(RPC_MAX_THREADS),
+    Utils.newDaemonThreadFactory("RPC-Handler-%d"));
+    int [] portData=getPortNumberAndRange();
+    int startingPortNumber=portData[PortRangeSchema.START_PORT.ordinal()];
+    int endPort=portData[PortRangeSchema.END_PORT.ordinal()];
+    for(int tries = startingPortNumber-1 ; tries <= endPort ; tries++){
+      try {
+        startingPortNumber++;
+        this.channel = getChannel(startingPortNumber);
+        break;
+      }catch(BindException e){
+        LOG.warn("RPC not able to connect port "+ startingPortNumber);
+      }
+    }
```

Comment:
Have Changed the code , to handle the case if no available port . It will throws exception and fails gracefully. 

Please let me know , if that ok .